src/packet/bwe/arrival_group.rs

```diff
@@ -0,0 +1,309 @@
+use std::mem;
+use std::time::{Duration, Instant};
+
+use crate::rtp::SeqNo;
+
+use super::AckedPacket;
+
+const BURST_TIME_INTERVAL: Duration = Duration::from_millis(5);
+
+#[derive(Debug, Default)]
+pub struct ArrivalGroup {
+    first_seq_no: Option<SeqNo>,
+    last_seq_no: Option<SeqNo>,
+    first_local_send_time: Option<Instant>,
+    local_send_time: Option<Instant>,
+    remote_recv_time: Option<Instant>,
+    size: usize,
+}
+
+impl ArrivalGroup {
+    /// Maybe add a packet to the group.
+    ///
+    /// ## Return
+    ///
+    /// * [`Belongs::Belongs`] if the packet belongs to the group and was added.
+    /// * [`Belongs::NewGroup`] if the packet is the start of a new group and thus wasn't added.
+    ///                         The caller should create a new group and add this packet to the
+    ///                         group.
+    /// * [`Belongs::Skipped`] if the packet was skipped i.e. because it was out of order.
+    fn add_packet(&mut self, packet: AckedPacket) -> Belongs {
+        let belongs = self.belongs_to_group(&packet);
+
+        if belongs == Belongs::NewGroup || belongs == Belongs::Skipped {
+            return belongs;
+        }
+
+        self.remote_recv_time = Some(packet.remote_recv_time);
```

Comment:
I think it would make sense to rename remote_recv_time to indicate it's the latest remove recv time.    last_remote_recv_time?

Along with that, you are making sure that these is always the largest/latest time value even if the packets are added out of order?

Reply:
Hmm, I had this before but I think I might have regressed it in some iteration of the code.

Reply:
The code that checks group membership already excludes packets that arrive out of order so there's no need to take the max here. I reviewed this code against libWebRTC and it now matches what they do